internal/engine/protocol/activitypub/data_source.go

```diff
@@ -42,173 +43,248 @@ func (s *dataSource) State() json.RawMessage {
 	return lo.Must(json.Marshal(s.state))
 }
 
-// Start initializes the data protocol and starts consuming Kafka messages
+// Start initializes the data protocol and starts consuming relay messages
 func (s *dataSource) Start(ctx context.Context, tasksChan chan<- *engine.Tasks, errorChan chan<- error) {
-	if err := s.initialize(); err != nil {
-		errorChan <- fmt.Errorf("initialize dataSource: %w", err)
+	if err := s.initialize(ctx, errorChan); err != nil {
+		zap.L().Error("Failed to initialize", zap.Error(err))
+		errorChan <- fmt.Errorf("failed to initialize dataSource: %w", err)
+
+		return
+	}
+
+	if s.mastodonClient == nil {
+		zap.L().Error("Mastodon client is nil after initialization")
 		return
 	}
 
-	// Start consuming Kafka messages in a goroutine
+	// Get the channel to get the notification of the http server being ready
+	readyChan := s.mastodonClient.GetReadyChan()
+
+	// Start processing in a goroutine
+	var wg sync.WaitGroup
+
+	wg.Add(1)
```

Comment:
You use `wg.Add(1)` but do not call `wg.Wait()`. This makes the WaitGroup redundant. You should either add a `wg.Wait()` after starting the goroutine to properly wait for it, or remove the WaitGroup entirely if it is not needed.

Reply:
added missing wg.Wait() at the end of the Start function.